linked_list/main.py

```diff
@@ -62,34 +66,129 @@ class LinkedList(AbstractLinkedList):
     """
 
     def __init__(self, elements=None):
-        pass
+        self.start = Node(elem=None)
+        self.end = self.start
+        self.counter = 0
+        self.i = self.start
+        if elements is not None:
+            for elem in elements:
+                if self.end.elem == None:
+                    self.end.elem = elem
+                else:
+                    self.end.next = Node(elem=elem)
+                    self.end = self.end.next
 
     def __str__(self):
-        pass
+        return str([ x for x in self ])
 
     def __len__(self):
-        pass
+        return self.count()
 
     def __iter__(self):
-        pass
+        return self
+
+    def __next__(self):
+        if self.counter < self.count():
+            elem = self.i.elem
+            self.counter += 1
+            self.i = self.i.next
+            return elem
+        else:
+            self.counter = 0
+            self.i = self.start
+            raise StopIteration
```

Comment:
So you mean to say that once the iter is exhausted, __iter__ is called? 

I first had self.counter and self.i in __iter__ but it was not working. Then moved it to __init__. But after doing that, if I called the iterator a second time, then it didn't returned anything. Because the counter variable was already incremented to len(obj). So had to clean up in the else part. 

I'll try immediately by moving the part in the __iter__